app/controllers/users_controller.rb

```diff
@@ -155,4 +156,16 @@ def login_user(user)
   def parameter_filter_context
     Foreman::Controller::Parameters::User::Context.new(:ui, controller_name, params[:action], editing_self?)
   end
+
+  def verify_active_session
+    if !request.post? && params[:status].blank? && User.exists?(session[:user].presence)
+      warning _("You have already logged in")
+      redirect_to back_or_hosts
```

Comment:
There is already a `redirect_back_or_to` helper in the application controller, you should be able to call `redirect_back_or_to hosts_path`.